pennylane/transforms/classical_jacobian.py

```diff
@@ -65,39 +72,126 @@ def classical_jacobian(qnode):
+
+    **Example with ``argnums``**
+
+    >>> @qml.qnode(dev)
+    ... def circuit(x, y, z):
+    ...     qml.RX(qml.math.sin(x), wires=0)
+    ...     qml.CNOT(wires=[0, 1])
+    ...     qml.RY(y ** 2, wires=1)
+    ...     qml.RZ(1 / z, wires=1)
+    ...     return qml.expval(qml.PauliZ(0)@qml.PauliZ(1))
+    >>> jac_fn = qml.transforms.classical_jacobian(circuit, argnums=[1, 2])
+    >>> x, y, z = np.array([0.1, -2.5, 0.71])
+    >>> jac_fn(x, y, z)
+    (array([-0., -5., -0.]), array([-0.        , -0.        , -1.98373339]))
+
+    Only the Jacobians with respect to the arguments ``x`` and ``y`` were computed, and
+    returned as a tuple of ``arrays``.
+
     """
 
     def classical_preprocessing(*args, **kwargs):
-        """Returns the trainable gate parameters for
-        a given QNode input"""
+        """Returns the trainable gate parameters for a given QNode input."""
         qnode.construct(args, kwargs)
         return qml.math.stack(qnode.qtape.get_parameters())
 
     if qnode.interface == "autograd":
-        return qml.jacobian(classical_preprocessing)
+
+        def _jacobian(*args, **kwargs):
+            if argnums is None:
+                jac = qml.jacobian(classical_preprocessing)(*args, **kwargs)
+            elif np.isscalar(argnums):
+                jac = qml.jacobian(classical_preprocessing, argnum=argnums)(*args, **kwargs)
+            else:
+                jac = tuple(
+                    (
+                        qml.jacobian(classical_preprocessing, argnum=i)(*args, **kwargs)
+                        for i in argnums
+                    )
+                )
+            return jac
+
+        return _jacobian
 
     if qnode.interface == "torch":
         import torch
 
         def _jacobian(*args, **kwargs):  # pylint: disable=unused-argument
-            return torch.autograd.functional.jacobian(classical_preprocessing, args)
+            jac = torch.autograd.functional.jacobian(classical_preprocessing, args)
+            if argnums is not None:
+                if np.isscalar(argnums):
+                    jac = jac[argnums]
+                else:
+                    jac = tuple((jac[idx] for idx in argnums))
+            return jac
 
         return _jacobian
 
     if qnode.interface == "jax":
         import jax
 
-        return jax.jacobian(classical_preprocessing)
+        argnums = 0 if argnums is None else argnums
+
+        def _jacobian(*args, **kwargs):
+            return jax.jacobian(classical_preprocessing, argnums=argnums)(*args, **kwargs)
+
+        return _jacobian
 
     if qnode.interface == "tf":
         import tensorflow as tf
 
         def _jacobian(*args, **kwargs):
+            if np.isscalar(argnums):
+                sub_args = args[argnums]
+            elif argnums is None:
+                sub_args = args
+            else:
+                sub_args = tuple((args[i] for i in argnums))
+
             with tf.GradientTape() as tape:
+                tape.watch(args)

```

Comment:
Oh, I'm curious how come you added this back in 🙂 

I had to remove this last week in a PR, since it was leading to errors with the TensorFlow backend --- `tape.watch(args)` was forcing _all_ QNode arguments to become trainable, even if a user had specified one of the args as `tf.constant(..)`.

This resulted in the `classical jacobian @ quantum jacobian` computation failing in the TF interface, since they would have mis-matching shapes

Reply:
Fixed it, thanks! Had borrowed this from before :)